frontend/src/metabase/reference/components/EditableReferenceHeader.jsx

```diff
@@ -0,0 +1,134 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Link } from "react-router";
+import cx from "classnames";
+import pure from "recompose/pure";
+
+import S from "./ReferenceHeader.css";
+import L from "metabase/components/List.css";
+import E from "metabase/reference/components/EditButton.css";
+
+import IconBorder from "metabase/components/IconBorder.jsx";
+import Icon from "metabase/components/Icon.jsx";
+import Ellipsified from "metabase/components/Ellipsified.jsx";
+import EditButton from "metabase/reference/components/EditButton.jsx";
+
+
+const EditableReferenceHeader = ({
+    entity = {},
+    table,
+    type,
+    headerIcon,
+    headerLink,
+    name,
+    user,
+    isEditing,
+    hasSingleSchema,
+    hasDisplayName,
+    startEditing,
+    displayNameFormField,
+    nameFormField
+}) =>
+    <div className="wrapper wrapper--trim">
+        <div className={cx("relative", L.header)} style={type === 'segment' ? {marginBottom: 0} : {}}>
+            <div className={L.leftIcons}>
+                { headerIcon &&
+                    <IconBorder
+                        borderWidth="0"
+                        style={{backgroundColor: "#E9F4F8"}}
+                    >
+                        <Icon
+                            className="text-brand"
+                            name={headerIcon}
+                            width={24}
+                            height={24}
+                        />
+                    </IconBorder>
+                }
+            </div>
+            { type === 'table' && !hasSingleSchema && !isEditing &&
+                <div className={S.headerSchema}>{entity.schema}</div>
+            }
+            <div
+                className={S.headerBody}
+                style={isEditing && name === 'Details' ? {alignItems: "flex-start"} : {}}
+            >
+                { isEditing && name === 'Details' ?
+                    hasDisplayName ?
```

Comment:
FWIW nested ternaries are great if you indent them properly. Then they become sort of like a `case` statement in `Clojure`. For example here's a bad FizzBuzz that at least has a nice ternary statement

```c
#include <stdio.h>

int main() {
  for (int i = 0; i < 100; i++) {
     printf((!(i % 15) ? "%d FizzBuzz\n" :
             !(i % 3)  ? "%d Fizz\n" :
             !(i % 5)  ? "%d Buzz\n" :
                         "%d\n"),
            i);
  }
  return 0;
}
```

it's preferable to a bunch of `if`s and `else`s I think